meshtastic/__main__.py

```
@@ -1371,11 +1502,23 @@ def common():
             elif args.host:
                 try:
                     if ":" in args.host:
-                        tcp_hostname, tcp_port = args.host.split(':')
+                        tcp_hostname, tcp_port_str = args.host.rsplit(":", 1)
+                        # Handle bracketed IPv6 addresses like [::1]:4403
+                        if tcp_hostname.startswith("[") and tcp_hostname.endswith("]"):
+                            tcp_hostname = tcp_hostname[1:-1]
+                        try:
+                            tcp_port = int(tcp_port_str)
+                            if not 1 <= tcp_port <= 65535:
+                                raise ValueError(f"Port {tcp_port} out of range")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)